geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/Stat.scala

```diff
@@ -79,7 +82,14 @@ trait Stat {
    *
    * @return stat as a json string
    */
-  def toJson: String
+  def toJson: String = Stat.JSON.toJson(toJsonObject)
+
+  /**
+    * Returns a representation of the stat to be serialized
+    *
+    * @return stat as a json serializable object
+    */
+  def toJsonObject: Any
```

Comment:
Yeah, it could be a union of {primitives( and boxed types), Date, and String; Maps and Seq containing previous type or themselves}. Thoughts on trying to express that?  In reality Any is appropriate as long as there's a registered serializer.